### PR TITLE
[Android] Fix casting of Location.getCurrentPositionAsync#timeout param

### LINKED ISSUE #2226: Location.getCurrentPositionAsync `timeout` option immediately throws an exception
### Environment

  OS: Linux 4.18
  Node: 8.12.0
  Yarn: 1.9.4
  npm: 6.4.1
  Watchman: Not Found
  Xcode: N/A
  Android Studio: Not Found

Packages: (wanted => installed)
  expo: ^30.0.0 => 30.0.1
  react: 16.3.1 => 16.3.1
  react-native: https://github.com/expo/react-native/archive/sdk-30.0.0.tar.gz => 0.55.4

Android Client, Standalone

### Steps to Reproduce

Simply try to call the getCurrentLocationAsync method on an Android device. I tested it on 5.0 and 7.0 (tablet).

### Expected Behavior

Return with the current position or timeout.

### Actual Behavior

Immediately throws an exception.

```
Encountered an exception while calling native method: Exception occurred while executing exported method getCurrentPositionAsync on module ExpoLocation: null" in c@/data/user/0/host.exp.exponent/files/30.0.0/cached-bundle-experience-%40snack%2Fsdk.30.0.0-HJqtWtyYm-1841382398-30.0.0:19:1623 << /data/user/0/host.exp.exponent/files/30.0.0/cached-bundle-experience-%40snack%2Fsdk.30.0.0-HJqtWtyYm-1841382398-30.0.0:19:990 << value@/data/user/0/host.exp.exponent/files/30.0.0/cached-bundle-experience-%40snack%2Fsdk.30.0.0-HJqtWtyYm-1841382398-30.0.0:21:3873 << /data/user/0/host.exp.exponent/files/30.0.0/cached-bundle-experience-%40snack%2Fsdk.30.0.0-HJqtWtyYm-1841382398-30.0.0:21:1397
```

### Reproducible Demo

[Snack](https://snack.expo.io/H14xAI0dQ)

### PR DESCRIPTION
**Pull request by bbarthec**
# Why

Resolves #2226

# How

Fixed casting issue

### CODE REVIEW

File: modules/expo-location/src/Location.js [hunk 1]
```diff
@@ -68,7 +68,12 @@ function getProviderStatusAsync(): Promise<ProviderStatus> {
   return Location.getProviderStatusAsync();
 }
 
-async function getCurrentPositionAsync(options: LocationOptions = {}): Promise<LocationData> {
+async function getCurrentPositionAsync(options: {
+  timeout?: number,
+  enableHighAccuracy?: boolean,
+  maximumAge?: number,
```
Comment (tsapeta) [inline modules/expo-location/src/Location.js]: why not as `LocationOptions` type?

**Review by tsapeta — CHANGES_REQUESTED** [review]
Looks good, but what about this comment https://github.com/expo/expo/issues/2226#issuecomment-427272746 ?

File: modules/expo-location/src/Location.js (hunk 1 above)
Comment (bbarthec) [reply modules/expo-location/src/Location.js]: I got no idea what is causing https://github.com/expo/expo/issues/2226#issuecomment-427272746 . More specific info is needed here as I've never encountered such problem on testing devices.

**Comment by bbarthec** [comment]
I got no idea what is causing https://github.com/expo/expo/issues/2226#issuecomment-427272746. More specific info is needed here as I've never encountered such problem on testing devices.

File: modules/expo-location/src/Location.js (hunk 1 above)
Comment (bbarthec) [reply modules/expo-location/src/Location.js]: I've changed this, because I've added this `timeout` option, but I'll reverse it

**Review by esamelson — APPROVED** [review]
This looks good! It seems like https://github.com/expo/expo/issues/2226#issuecomment-427272746 is a separate issue (doesn't involve `timeout`) so let's not worry about it for this PR.

Reminder to add `timeout` to docs :)

**Pull request merged**